scrapy_splash/middleware.py

```diff
@@ -397,19 +398,24 @@ def process_response(self, request, response, spider):
         return response
 
     def _change_response_class(self, request, response):
-        from scrapy_splash import SplashResponse, SplashTextResponse
-        if not isinstance(response, (SplashResponse, SplashTextResponse)):
+        from scrapy_splash import SplashResponse
+        from scrapy_splash.response import splash_scrapy_text_responses
+        splash_text_response_types = tuple(x for x, y in splash_scrapy_text_responses)
+
+        if not isinstance(response, (SplashResponse, splash_text_response_types)):
```

Comment:
Your new `SplashHtmlResponse` class inherits from `SplashTextResponse`. Doesn’t that make these changes and some changes below unnecessary?